hazelcast/src/main/java/com/hazelcast/client/impl/protocol/template/ResponseTemplate.java

```diff
@@ -77,6 +78,12 @@
     void ListData(List<Data> response);
 
     /**
+     * @param response The operation result as an array of serialized byte-array that might have null entries.
+     */
+    @Response(ResponseMessageConst.LIST_DATA_MAYBE_NULL_ELEMENTS)
```

Comment:
I believe its better not to modify existing LIST_DATA since we can not guarantee that we do not break existing contracts. In my opinion a new response type seemed a better fit to differentiate between the two.